test/metabase/api/database_test.clj

```diff
@@ -281,6 +282,19 @@
            (let [resp ((mt/user->client :rasta) :get 200 (format "database/%d/metadata" (mt/id)))]
              (assoc resp :tables (filter #(= "CATEGORIES" (:name %)) (:tables resp))))))))
 
+(defn- is-named [name]
+  #(= name (:name %)))
+
+(deftest fetch-database-metadata-include-hidden-test
+  (mt/with-temp-vals-in-db Table (mt/id :categories) {:visibility_type "hidden"}
+    (testing "GET /api/database/:id/metadata?include_hidden_tables=true"
+      (let [tables (:tables ((mt/user->client :rasta) :get 200 (format "database/%d/metadata?include_hidden_tables=true" (mt/id))))]
+        (is (some (is-named "CATEGORIES") tables))))
```

Comment:
This doesn't matter too much but you could do something like 

```clj
(some (partial = "CATEGORIES") (map :name tables))
```

or
 
```clj
(contains? (set (map :name tables)) "CATEGORIES")
```

instead of defining the separate function